### ISSUE TITLE
`role-suports-aria-props` is not flagging prohibited accessible name usage on generic elements

### ISSUE THREAD

**Issue by khiga8** [jsx-eslint/eslint-plugin-jsx-a11y#910]
I would expect [role-suports-aria-props](https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/blob/main/docs/rules/role-supports-aria-props.md)  to flag [prohibited accessible name usage](https://w3c.github.io/aria/#namefromprohibited) on generic elements like:

```
<div aria-label="Some label">
<span aria-label="Some label">
<div aria-labelledby="some-id">
<span aria-labelledby="some-id">
```

etc.

We see this misuse of accessible name a lot in our projects and we're hoping this is a case we can cover with `eslint-plugin-jsx-a11y`.

I'd hoped that the latest bump of aria-query in https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/pull/814 would start flagging these issues, but when I update the tests in `__tests__/src/rules/role-supports-aria-props-test.js` on the latest `main` to include invalid cases like:

```
    {
      code: '<div aria-label />',
      errors: [errorMessage('aria-label', 'generic', 'div', true)],
    },
```

it doesn't seem like it's being recognized as invalid.

**Comment by jessebeach** [comment]
@khiga8 hello Kate! Directionally, you are on to something here. We don't have a rule that prohibits labels on elements or in combination with roles that do not support a label. I wouldn't do that in this rule, because it is specifically concerned with the combination of ARIA Role and ARIA Prop. In your examples, none of the HTML elements have a role. It's true that the `<div>` element corresponds to the `generic` role, so a label should be disallowed in this case. But what I think we need here is a new rule altogether.

**Comment by jessebeach** [comment]
@khiga8 would you like to update this issue to reflect that you're asking for a new rule?

**Comment by khiga8** [comment]
Hi @jessebeach! I'd noticed in the `role-supports-aria-props-test`, there are [checks for implicit roles](https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/blob/main/__tests__/src/rules/role-supports-aria-props-test.js#L88) and the [doc](https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/blob/main/docs/rules/role-supports-aria-props.md#jsx-a11yrole-supports-aria-props) describes the rule as the following:

> Enforce that elements with explicit or implicit roles defined contain only aria-* properties supported by that role.

I'm not sure I understand quite why this use case falls outside of this rule. Would you mind elaborating 🙏?

**Comment by jessebeach** [comment]
> Hi @jessebeach! I'd noticed in the `role-supports-aria-props-test`, there are [checks for implicit roles](https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/blob/main/__tests__/src/rules/role-supports-aria-props-test.js#L88) and the [doc](https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/blob/main/docs/rules/role-supports-aria-props.md#jsx-a11yrole-supports-aria-props) describes the rule as the following:
> 
> > Enforce that elements with explicit or implicit roles defined contain only aria-* properties supported by that role.
> 
> I'm not sure I understand quite why this use case falls outside of this rule. Would you mind elaborating 🙏?

You're right! Alright, this is the right place for this sort of restriction and I agree with you now that this rule isn't catching all cases. `aria-label` is marked as prohibited for the `generic` role: https://github.com/A11yance/aria-query/blob/main/src/etc/roles/literal/genericRole.js#LL12

The best way to approach this is for me to update `aria-query` with the allowed and prohibited aria props information, then reference that in this rule. I'll put it on the to-do list! 

Thank you for being persistent here!

**Comment by smockle** [comment]
I opened a PR to fix this: https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/pull/911

**Comment by drwpow** [comment]
We’ve had similar issues and needed to supplement jsx-a11y with our own rules that catch this. I had some time recently and created [html-aria](https://www.npmjs.com/package/html-aria), which extends basic aria-query with the [HTML in ARIA](https://www.w3.org/TR/html-aria/) normative recommendations that provide more accurate recommendations for applying ARIA roles to HTML, in addition to handling the `prohibited` attributes mentioned in this issue. At the moment it‘s in beta, and only supports ARIA 1.3 which would have some minor breaking changes for this library. But @jessebeach in case `html-aria` would be quicker to solve this than updating aria-query I’d be happy to collaborate / fix any bugs you come across 🙂
